src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -752,7 +752,11 @@ const StudentApplication = ({ userProfile, authentication }) => {
                   !applicationCardsOpen ||
                   !agreements ||
                   !(applicationDetails?.Applicants?.length > 0) ||
-                  !(applicationDetails?.ApartmentChoices?.length > 0)
+                  !(
+                    applicationDetails?.ApartmentChoices?.filter(
+                      (apartmentChoice) => apartmentChoice.HallName,
+                    )?.length > 0
+                  )
```

Comment:
Could you explain what this was doing before vs now?

Reply:
Oh, that did save. Weird. Well, you do not have to answer the question twice.

Reply:
Before, it was a check for whether the array of ApartmentChoices was empty.

Now, it uses the `.filter` to get the subset of array which have a nonempty HallName, and checks that
In other words, `ApartmentChoices.filter((apartmentChoice) => apartmentChoice.HallName)` returns an array of ApartmentChoices containing only the elements that have a real value selected for the `HallName`.

So basically, it filters out any of the Hall dropdown menus that are blank (i.e., the ones that don't have a hall selected yet), because before you could hit the "Add Hall" button but leave the dropdown blank, and it would still allow you to submit press submit even though there were technically no halls selected. 

Additionally, using the `?.` (such as in `ApartmentChoices?.filter()?.length` ) means that if the array itself is null, then it shortcuts the thing after the `?.` and returns null, instead of trying to perform `null.length` which would give an error. And then the `null > 0` returns false, so the `array?.length > 0` returns false if the array is empty or if the array variable is null, so we get the desired behavior of `(array?.length > 0) === true` whenever the array is not null and not empty.

Reply:
Okay, sweet, that makes plenty of sense.